buildSrc/src/main/java/org/elasticsearch/gradle/plugin/PluginPropertiesExtension.java

```
@@ -0,0 +1,164 @@
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getClassname() {
+        return classname;
+    }
+
+    public void setClassname(String classname) {
+        this.classname = classname;
+    }
+
+    public Collection<String> getExtendedPlugins() {
+        return Collections.unmodifiableCollection(extendedPlugins);
+    }
+
+    public boolean isHasNativeController() {
+        return hasNativeController;
+    }
+
+    public void setHasNativeController(boolean hasNativeController) {
+        this.hasNativeController = hasNativeController;
+    }
+
+    public boolean isHasClientJar() {
+        return hasClientJar;
+    }
+
+    public void setHasClientJar(boolean hasClientJar) {
+        this.hasClientJar = hasClientJar;
+    }
+
+    public boolean isRequiresKeystore() {
+        return requiresKeystore;
+    }
+
+    public void setRequiresKeystore(boolean requiresKeystore) {
+        this.requiresKeystore = requiresKeystore;
+    }
+
+    public File getLicenseFile() {
+        return licenseFile;
+    }
+
+    public void setLicenseFile(File licenseFile) {
+        this.project.getExtensions().add("licenseFile", licenseFile);
+        this.licenseFile = licenseFile;
+    }
+
+    public File getNoticeFile() {
+        return noticeFile;
+    }
+
+    public void setNoticeFile(File noticeFile) {
+        this.project.getExtensions().add("noticeFile", noticeFile);
+        this.noticeFile = noticeFile;
+    }
+
+    public Project getProject() {
+        return project;
+    }
+
+    public void setExtendedPlugins(List<String> extendedPlugins) {
+        if(extendedPlugins != null) {
```

Why this leniency? If we set extendedPlugins, then it should be set. This would result in very difficult to debug behavior if someone extendedPlugins was set once before.

---

Please check this review comment by @atorok  https://github.com/elastic/elasticsearch/pull/39605#discussion_r262972790 (recommended adding a null-check)

---

I think a null check is incorrect. As I noted above, it means it is "set once", but silently skips future sets.

---

Make sense! I will drop the null-check.

---

This is not what I had in mind with the null check. 
We should fail fast if the extenedPlugins is being set to a `null`. 
It's sometimes easy to set it to something in a build script that can return a null, we should should fail the build right away by throwing an exception as it eases debugging and makes it obvious where the null came from.